peps/pep-0727.rst

```diff
@@ -67,278 +81,438 @@ Rationale
+that support it.
 
+Library authors that wish to adopt this proposal while keeping backwards compatibility
+with older versions of Python should use ``typing_extensions.Doc`` instead of
+``typing.Doc``.
 
-Implementers can consider this invalid and are not required to support this to be
-considered conformant.
 
-Nevertheless, as it might be difficult to enforce it on developers, implementers
-can opt to support one of the ``doc()`` declarations.
+Reference Implementation
+========================
 
-In that case, the suggestion would be to support the last one, just because
-this would support overriding, for example, in:
+``typing.Doc`` is implemented equivalently to:
 
 .. code:: python
 
-   User = Annotated[str, doc("A user name")]
-
-   CurrentUser = Annotated[User, doc("The current user name")]
+   class Doc:
+       def __init__(self, documentation: str, /):
+           self.documentation = documentation
 
 
-Internally, in Python, ``CurrentUser`` here is equivalent to:
+It has been implemented in the `typing_extensions`__ package.
 
-.. code:: python
+__ https://pypi.org/project/typing-extensions/
 
-   CurrentUser = Annotated[str,
-                           doc("A user name"),
-                           doc("The current user name")]
 
+Survey of Other languages
+=========================
 
-For an implementation that supports the last ``doc()`` appearance, the above
-example would be equivalent to:
+Here's a short survey of how other languages document their symbols.
 
-.. code:: python
 
-   def hi(to: Annotated[str, doc("The current user name")]) -> None: ...
+Java
+----
 
+Java functions and their parameters are documented with
+`Javadoc <https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html>`__,
+a special format for comments put on top of the function definition. This would be
+similar to Python current docstring microsyntax conventions (but only one).
 
-.. you need to fill these in:
+For example:
 
-   Backwards Compatibility
-   =======================
+.. code:: java
 
-   [Describe potential impact and severity on pre-existing code.]
+   /**
+   * Returns an Image object that can then be painted on the screen. 
+   * The url argument must specify an absolute <a href="#{@link}">{@link URL}</a>. The name
+   * argument is a specifier that is relative to the url argument. 
+   * <p>
+   * This method always returns immediately, whether or not the 
+   * image exists. When this applet attempts to draw the image on
+   * the screen, the data will be loaded. The graphics primitives 
+   * that draw the image will incrementally paint on the screen. 
+   *
+   * @param  url  an absolute URL giving the base location of the image
+   * @param  name the location of the image, relative to the url argument
+   * @return      the image at the specified URL
+   * @see         Image
+   */
+   public Image getImage(URL url, String name) {
+   try {
```

Comment:
Can you add indentation inside the function?

Reply:
Sure thing! I copy-pasted it from the Java docs as-is, but it makes more sense to have some indentation. Done! ✅